### ISSUE TITLE
Position information is not reported when arranging start menu tiles with alt+shift+arrow

### ISSUE THREAD

**Issue by cary-rowen** [josephsl/wintenApps#69]
In the readme there is the following text:

>
* When arranging Start menu tiles or Action Center quick actions with Alt+Shift+arrow keys, NVDA will announce information on dragged items or new position of the dragged item.

## Actual behavior
NVDA is silent when trying to arrange tile positions with alt+Shift+arrows.

## Expected behavior
nvda reports moved location information.

## other information

* NVDA2021.3.3
* Windows 10 21H2 (x64) build 19044.1526
* Narrator works as expected

**Comment by josephsl** [comment]
Hi, is “cell coordinates” checked under Document Formatting/tables group? On my computer, Narrator doesn’t report the new position of the dragged tile – it announces the name of the tile and the fact that drag is complete, whereas NVDA does announce the new coordinate (if the mentioned setting is turned on). Thanks.

**Comment by cary-rowen** [comment]
Yes as you said, with this setting turned on NVDA will report the new location and tile name.
But narrator's user experience could be better because narrator announces where to move to, as well as drag and drop or c create new groups during the move.
However, nvda only reads out the new location and name when the tile is down, when I hold alt+shift and the four arrow keys to move it doesn't announce anything, only when I release those keys it will Report new location and tile name, So much so that I have no idea where the tile should be placed.
Do you have any plans to improve this?

**Comment by josephsl** [comment]
Hi, what you hear from Narrator is a help message of some sort, and the message you hear about dragging items come from an event the add-on does not track. As for supporting these, I don’t think so at this time. Thanks.

**Comment by josephsl** [comment]
Hi,

Technical: NVDA must track drag start/complete/drop target events (at a minimum) to provide messages similar to what Narrator announces. Older add-on releases tracked additional drag/drop target events but were removed as they do not provide information such as drag direction and context information (unless I didn't read UIA documentation thoroughly).

While NVDA is aligned a bit closely to Narrator's behavior and messages, these screen readers are fundamentally different. There is also the question of balancing between verbosity and efficiency in circumstances like this, and I (personally) tend to favor efficiency a bit more than verbosity unless the situation really warrants it.

If users believe the suggestion outlined above should be implemented, then feel free to add additional comments and justifications, or if you are willing, provide a pull request.

Thanks.

**Comment by josephsl** [comment]
Hi,

Additionally, while I do plan to support Windows 10 until Microsoft ends support, it is highly unlikely that major features will be implemented in Windows 10 by Microsoft. It is also likely that, as far as add-on features are concerned, more features will be made available for Windows 11, and I will do my best to provide solutions for Windows 10 if possible (may not be identical, but perhaps equivalent).

In case of this issue, while I have no solutions at the moment, it might be possible that others might.

Thanks.

**Comment by cary-rowen** [comment]
HI Joseph,

Thank you for your kind reply,
As far as efficiency is concerned, I think it will definitely improve efficiency by reporting the passed position from time to time during the movement, otherwise, the user will need to put the tile down to know where he is.
In this case if that location is not what they want, they have to drag and drop again.

Grateful

**Comment by josephsl** [comment]
Hi, testing confirmed what I thought is happening: an event NVDA is not tracking at the moment (need to confirm if it did use to track this but was dropped). A possible solution will be deployed within the next hour or so. Thanks.

**Comment by josephsl** [comment]
Hi,

Technical: NVDA is not tracking drag drop effect property. This means a new property change event must be defined (in this case, inserted i.e. a new event must be added when the global plugin portion loads), either given a new UIA property event name or assigned to an existing event. This change may or may not become part of the upcoming stable version of the add-on (depends on when a prototype is released so it can be tested in a variety of scenarios); at the moment it is highly unlikely that it will be part of the next stable version (March).

Thanks.

**Comment by josephsl** [comment]
Hi,

Look out for Windows App Essentials February 25th development build or later.

Thanks.

**Comment by cary-rowen** [comment]
Hi @josephsl 

Thanks for investigating the issue and giving such a specific explanation👍👍
I'll keep an eye on the development version or later.

Grateful

**Comment by josephsl** [comment]
Hi,

In case folks didn't getthe February 25th dev build, it can be downloaded from here:

https://github.com/josephsl/wintenApps/releases/download/22.02/wintenApps-20220225-dev.nvda-addon

Thanks.

**Comment by josephsl** [comment]
Hi, I’ll close this issue as resolved in a few days. Thanks.

**Comment by akash07k** [comment]
hey @josephsl 
Thanks for fixing it, but still the position coordinates/information is not being announced while rearranging the items in the taskbar. (I know that it is not mentioned in the readme, but thought to mention it anyways)

**Comment by josephsl** [comment]
Hi, yep, I did notice this, and testing with both NVDA and Narrator tells me that Narrator is not announcing this info either (at least in Windows 11) – may need to test this in Windows 10 at some point. Thanks.

**Comment by akash07k** [comment]
Ok

**Comment by cary-rowen** [comment]
Hi,

According to my tests, the Windows 10 taskbar does not support the use of Alt+Shift and arrow keys to move the icon's position.
Windows 10 21H2 (x64) build 19044.1526

Thanks

**Comment by josephsl** [comment]
Hi, so at least this confirms part of what I suspected – thanks for this report. It looks like taskbar arrangement announcement is a bit beyond the scope of this add-on for now. Thanks.

**Comment by cary-rowen** [comment]
Hi, 

Windows 11 can use Alt+Shift+Left and Right arrows to move the position of the taskbar icon, but neither Narrator nor NVDA can report the relevant location information.

Windows 11 21H2 (x64) build 22000.527

Thanks
